### ISSUE TITLE
android: fix linking issue post bazel 1.2.0

### ISSUE THREAD

**Issue by junr03** [lyft/envoy-mobile#604]
The app cannot find the symbols for the jni interface functions if rules depend solely on a `cc_libbrary` target. For the time being a `cc_binary` target was used with linkshared true in #600. Further investigation is needed.

**Comment by junr03** [comment]
Additionally the native dependency was added to the top level aar rather than the specific targets that require it. Overall @buildbreaker and my opinion is that the android rules could be ironed out a bit.

**Comment by junr03** [comment]
not stale

**Comment by arturdryomov** [comment]
It might be related to bazelbuild/bazel#10051 ([which is a bit weird](https://github.com/bazelbuild/bazel/issues/10051#issuecomment-565123413)).

**Comment by buildbreaker** [comment]
@arturdryomov Yeah, that does sound related 😬 . Seems like you have linked a number of issues related to the `incompatible_remove_legacy_whole_archive` issue. Hopefully we will hear about that sometime soon and we can have a more instructive approach to solving this problem on our end

**Comment by arturdryomov** [comment]
@buildbreaker, for now I believe it is possible to set `alwayslink` to `True` and it should do the trick.
